### PR TITLE
feat: setup rate limiting

### LINKED ISSUE #7: Setup rate limiting

### PR DESCRIPTION
**Pull request by maxyzli**
resolves #7

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -32,6 +34,17 @@ services:
...
+    container_name: redis
+    image: redis:alpine
+    command: redis-server --requirepass sOmE_sEcUrE_pAsS
+    ports:
+      - 6379:6379
+    environment:
+      - REDIS_REPLICATION_MODE=master
+    volumes:
+      - redis:/data
+
```
Comment (geoffturk) [inline docker-compose.dev.yml:47]: Is there a reason you did not add this to `docker-compose.production.yml` as well?

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -57,4 +70,5 @@ services:
 volumes:
   postgresql:
   mongodb:
+  redis:
```
Comment (geoffturk) [inline docker-compose.dev.yml:73]: Same here.

**Review by geoffturk — APPROVED** [review]
This works fine. See my comment about adding it to the production Dockerfile as well. Any reason for leaving it out?

**Comment by maxyzli** [comment]
@geoffturk You can update the production.yaml. I won't touch it.

**Pull request merged**